### ISSUE TITLE
Spring secret manager unable to retrieve secret when running in CloudBuild using cloud-builders and Dockerfile

### ISSUE THREAD

**Issue by mservidio** [GoogleCloudPlatform/spring-cloud-gcp#2075]
**Describe the bug**
We have a Java project that uses secrets stored in secret manager. Locally, we can successfully retrieve secrets when using the following in a service constructor:

```
public TestService(
            @Value("${sm://password_value}") final String password
) {
```

Builds are performed using cloud build, and if we use a cloudbuild.yaml as follows, builds work, and tests work as the secret values are injected.

```
steps:
  - name: gradle:7.6-jdk17-jammy
    id: Build
    dir: 'src'
    entrypoint: gradle
    args: [ "build" ]
```

However, when we use gcr.io/cloud-builders/docker instead and build using a Dockerfile that uses gradle:jdk17, the same spring boot profile that succeeds using gradle, fails when using the cloud-builders/Dockerfile approach with a timeout.

```
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     Caused by: com.google.api.gax.rpc.DeadlineExceededException: io.grpc.StatusRuntimeException: DEADLINE_EXCEEDED: deadline exceeded after 7.349322249s. [closed=[], open=[[buffered_nanos=7355943028, waiting_for_connection]]]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:94) ~[gax-2.31.1.jar:2.31.1]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:41) ~[gax-2.31.1.jar:2.31.1]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:86) ~[gax-grpc-2.31.0.jar:2.31.0]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:66) ~[gax-grpc-2.31.0.jar:2.31.0]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:97) ~[gax-grpc-2.31.0.jar:2.31.0]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:84) ~[api-common-2.14.1.jar:2.14.1]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1127) ~[guava-32.1.1-jre.jar:na]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:31) ~[guava-32.1.1-jre.jar:na]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1286) ~[guava-32.1.1-jre.jar:na]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:1055) ~[guava-32.1.1-jre.jar:na]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:807) ~[guava-32.1.1-jre.jar:na]
2023-07-28T02:15:28.916+0000 [DEBUG] [TestEventLogger]     	at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:574) ~[grpc-stub-1.55.3.jar:1.55.3]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:544) ~[grpc-stub-1.55.3.jar:1.55.3]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at io.grpc.PartialForwardingClientCallListener.onClose(PartialForwardingClientCallListener.java:39) ~[grpc-api-1.55.3.jar:1.55.3]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:23) ~[grpc-api-1.55.3.jar:1.55.3]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at io.grpc.ForwardingClientCallListener$SimpleForwardingClientCallListener.onClose(ForwardingClientCallListener.java:40) ~[grpc-api-1.55.3.jar:1.55.3]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at com.google.api.gax.grpc.ChannelPool$ReleasingClientCall$1.onClose(ChannelPool.java:541) ~[gax-grpc-2.31.0.jar:2.31.0]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:567) ~[grpc-core-1.55.3.jar:1.55.3]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:71) ~[grpc-core-1.55.3.jar:1.55.3]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:735) ~[grpc-core-1.55.3.jar:1.55.3]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:716) ~[grpc-core-1.55.3.jar:1.55.3]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.55.3.jar:1.55.3]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.55.3.jar:1.55.3]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
2023-07-28T02:15:28.917+0000 [DEBUG] [TestEventLogger]     	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
2023-07-28T02:15:28.918+0000 [DEBUG] [TestEventLogger]     	Suppressed: com.google.api.gax.rpc.AsyncTaskException: Asynchronous task failed
2023-07-28T02:15:28.918+0000 [DEBUG] [TestEventLogger]     		at com.google.api.gax.rpc.ApiExceptions.callAndTranslateApiException(ApiExceptions.java:57) ~[gax-2.31.1.jar:2.31.1]
2023-07-28T02:15:28.918+0000 [DEBUG] [TestEventLogger]     		at com.google.api.gax.rpc.UnaryCallable.call(UnaryCallable.java:112) ~[gax-2.31.1.jar:2.31.1]
2023-07-28T02:15:28.918+0000 [DEBUG] [TestEventLogger]     		at com.google.cloud.secretmanager.v1.SecretManagerServiceClient.accessSecretVersion(SecretManagerServiceClient.java:1396) ~[google-cloud-secretmanager-2.21.0.jar:2.21.0]
2023-07-28T02:15:28.918+0000 [DEBUG] [TestEventLogger]     		at com.google.cloud.secretmanager.v1.SecretManagerServiceClient.accessSecretVersion(SecretManagerServiceClient.java:1326) ~[google-cloud-secretmanager-2.21.0.jar:2.21.0]
2023-07-28T02:15:28.918+0000 [DEBUG] [TestEventLogger]     		at com.google.cloud.spring.secretmanager.SecretManagerTemplate.getSecretByteString(SecretManagerTemplate.java:193) ~[spring-cloud-gcp-secretmanager-4.6.0.jar:4.6.0]
2023-07-28T02:15:28.918+0000 [DEBUG] [TestEventLogger]     		at com.google.cloud.spring.secretmanager.SecretManagerPropertySource.getProperty(SecretManagerPropertySource.java:47) ~[spring-cloud-gcp-secretmanager-4.6.0.jar:4.6.0]
```

Is there some other configuration that we would need to make in the Dockerfile for this to work? Valid ADC credentials are already available at '/root/.config/gcloud/application_default_credentials.json' within the Dockerfile and are being used successfully to get authorized dependencies from artifact registry. Does SecretManager use a different location for ADC perhaps?

**Comment by meltsufin** [comment]
I think this is more of a question for the Cloud Build product. I would recommend contacting Google Cloud Support.

**Comment by mservidio** [comment]
@meltsufin I've worked around the issue by supplying the secrets through the cloud build file. It's not as concise as it would be if we could retrieve secrets directly from the spring boot application, but it does the job for now. But, I'd definitely like to get to the bottom of the issue so we can use the sm:// functionality.

Is there any specific logging I could activate to try and debug the issue?

```
steps:
  - name: gcr.io/cloud-builders/docker
    entrypoint: bash
    args:
      - "-c"
      - |
        docker build --no-cache --network=cloudbuild \
        --build-arg BUILD_VERSION=$_BUILD_VERSION \
        --build-arg _CACHE_AUTH_PASSWORD="$$CACHE_AUTH_PASSWORD" \
        --build-arg _CACHE_CERTIFICATE="$$CACHE_CERTIFICATE" \
        -t $_AR_HOSTNAME/$PROJECT_ID/cloud-run-source-deploy/$REPO_NAME/$_SERVICE_NAME:$COMMIT_SHA \
        src/web-api \
        -f src/web-api/Dockerfile
    id: Build
    secretEnv: ['CACHE_AUTH_PASSWORD', 'CACHE_CERTIFICATE']
```

**Comment by mservidio** [comment]
Ok, so I stumbled upon my mistake. My other project/repository that successfully connects to Artifactory Repository is on the [cloudbuild network](https://cloud.google.com/build/docs/build-config-file-schema#network) (--network=cloudbuild). When I implemented the work-around to supply the secrets through the cloud build yaml file, I switched the entrypoint to bash to allow for passing in the environmental variables, and I had copied the command from another project that I have that has success connecting to Artifactory (which has --network=cloudbuild).

Adding the cloudbuild network fixed the issue.
